server/neptune/gateway/event/preworkflow/hook_executor.go

```diff
@@ -0,0 +1,64 @@
+package preworkflow
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"os"
+	"os/exec"
+)
+
+// interface for actual execution makes local testing easier
+
+type HookExecutor interface {
+	Execute(hook *valid.PreWorkflowHook, repo models.Repo, path string) error
+}
+
+type PreWorkflowHookExecutor struct {
```

Comment:
nit: should we name this HookExecutor so that it can be imported as `preworkflow.HookExecutor`?

Reply:
I can't redeclare names in the same package (runner and executor are in the same package, so the interface already uses the name)

Reply:
update: I think I can get around it by instead calling it `HookExecutor` and just calling the interface `Executor`.  Fixed in follow-up commit.